### PR TITLE
change the used organization header

### LINKED ISSUE #642: Load wards by organization
### Describe the chore

Right now, the wards are loaded and edited in the context of [the hard-coded organization](https://github.com/helpwave/web/blob/main/tasks/utils/grpc.ts#L36). We should pass the required organization ID to `getAuthenticatedGrpcMetadata()`.

The ID of the parent organization gets already propagated to the Ward Overview at `/organizations/121d3f0f-779c-46cf-82a8-f6353aa6367b`.

### LINKED ISSUE #669: Stations are displayed although no station was created
### Describe the bug

Stations are displayed although no station was created.		

### Describe how to produce the bug

Create a new account on [production](https://tasks.helpwave.de).
Station are thus displayed, to which you have access, but should not have.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DasProffi**
## Which issues does this pull request close?
closes #642
closes #669

### CODE REVIEW

**Comment by DasProffi** [comment]
This creates an error currently, because the token claims aren't set properly by the backend

File: tasks/utils/grpc.ts [hunk 1]
```diff
@@ -28,11 +28,12 @@ type AuthenticatedGrpcMetadata = {
   'X-Organization': string
 }
 
-export const getAuthenticatedGrpcMetadata = (): AuthenticatedGrpcMetadata => {
+const defaultOrganization = `3b25c6f5-4705-4074-9fc6-a50c28eba406`
```
[the remark this answers is not in the document]
Comment (DasProffi) [reply tasks/utils/grpc.ts:31]: What do you mean?

[the remark this answers is not in the document]
Comment (DasProffi) [reply tasks/utils/grpc.ts:31]: The backend expects a header and not all request like bed create could provide an organisation id. Removing the default here would mean coming up with an organization in all requests or having a dependency to backend changes.
Maybe @MaxSchaefer can provide us with the current state of the X-Organisation header

[the remark this answers is not in the document]
Comment (MaxSchaefer) [reply tasks/utils/grpc.ts:31]: @DasProffi that's correct. Currently, the `X-Organization` header is required. https://github.com/helpwave/services/pull/374 is also about to introduce `3b25c6f5-4705-4074-9fc6-a50c28eba406` (demo org) as a "default"/"fallback" organization ID. As already (internal) mentioned, this behavior is of course a huge tech debt and will be addressed after a restructure of our multi-tenancy approach. cc. @helpwave/medical

[the remark this answers is not in the document]
Comment (MaxSchaefer) [reply tasks/utils/grpc.ts:31]: @DasProffi please make sure to add a `// todo`.

**Comment by MaxSchaefer** [comment]
@DasProffi lets hop on a call before merging.

File: tasks/utils/grpc.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (use-to) [reply tasks/utils/grpc.ts:31]: It's just not obvious I suggest the uuid to be the nil uuid as a default organization. These hard-coded random uuids are totally unreadable and we should document these internal communications more (mention it in the issue)

[the remark this answers is not in the document]
Comment (use-to) [reply tasks/utils/grpc.ts:31]: @MaxSchaefer A demo organization should not have the same uuid for everyone, right? So why should a default uuid be hardcoded?

**Comment by use-to** [comment]
> @DasProffi lets hop on a call before merging.

Yes, please. You should invite to that meeting.

File: tasks/components/layout/WardDetails.tsx [hunk 2]
```diff
@@ -95,11 +99,11 @@ export const WardDetail = ({
     }
   }, [data, isCreatingNewWard])
 
-  const createWardMutation = useWardCreateMutation((ward) => context.updateContext({ ...context.state, wardId: ward.id }))
-  const updateWardMutation = useWardUpdateMutation((ward) => {
+  const createWardMutation = useWardCreateMutation(organisationId, (ward) => context.updateContext({ ...context.state, wardId: ward.id }))
```
Comment (use-to) [inline tasks/components/layout/WardDetails.tsx:102]:
```suggestion
  const createWardMutation = useWardCreateMutation(organizationId, (ward) => context.updateContext({ ...context.state, wardId: ward.id }))
```

Comment (use-to) [reply tasks/components/layout/WardDetails.tsx:102]: We agreed some time ago on using the american english spelling.

File: tasks/utils/grpc.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (use-to) [reply tasks/utils/grpc.ts:31]:
```suggestion
const defaultOrganization = `00000000-0000-0000-0000-000000000000`
```

**Review by use-to — CHANGES_REQUESTED** [review]

**Comment by florianjs** [comment]
[Bildschirmaufzeichnung vom 2023-08-26, 12-28-09.webm](https://github.com/helpwave/web/assets/54689374/a665469b-842d-46dc-ae05-e4df82d433d3)

**Comment by MaxSchaefer** [comment]
> Bildschirmaufzeichnung.vom.2023-08-26.12-28-09.webm 
> we should check if the user has acces to the Organisation bevor showing

Until refresh tokens. That's a very good idea. I will add a modal that asks after "Accept invite" and "Create organization" to re-login to helpwave tasks (this should happen not often). The modal has the following choices, "Yes, logout." / "No, later." with the information that the new organizations will be visible in the UI after a re-login.

**Comment by MaxSchaefer** [comment]
Open to-dos
- [ ] Display "ReSignInModal" after "CreateOrganization" and "AcceptInvitation" @DasProffi 
- [ ] Hide organizations in the organizations listing that are not in the current access token @DasProffi 
- [ ] Perform merge with https://github.com/helpwave/services/pull/374 @MaxSchaefer

**Comment by MaxSchaefer** [comment]
@use-to you've requested changes :) Please take a look.

**Pull request merged**